Sources/Purchasing/Configuration.swift

```diff
@@ -0,0 +1,150 @@
+    let storeKit2Setting: StoreKit2Setting
+    let dangerousSettings: DangerousSettings?
+    let networkTimeoutSeconds: Int
+    let storeKit1TimeoutSeconds: Int
+
+    override public init() {
+        fatalError("Use static function configurationBuilder(withAPIKey:) to configure.")
+    }
+
+    private init(withConfigurationBuilder configurationBuilder: ConfigurationBuilder) {
+        self.apiKey = configurationBuilder.apiKey
+        self.appUserID = configurationBuilder.appUserID
+        self.observerMode = configurationBuilder.observerMode
+        self.userDefaults = configurationBuilder.userDefaults
+        self.storeKit2Setting = configurationBuilder.storeKit2Setting
+        self.dangerousSettings = configurationBuilder.dangerousSettings
+
+        if configurationBuilder.storeKit1TimeoutSeconds < 5 {
+            self.storeKit1TimeoutSeconds = 5
+        } else {
+            self.storeKit1TimeoutSeconds = configurationBuilder.storeKit1TimeoutSeconds
+        }
+
+        if configurationBuilder.networkTimeoutSeconds < 5 {
+            self.networkTimeoutSeconds = 5
+        } else {
+            self.networkTimeoutSeconds = configurationBuilder.networkTimeoutSeconds
+        }
+    }
+
+    /// Factory method for the ``ConfigurationBuilder`` object that is required to create a `Configuration`
+    @objc public static func configurationBuilder(withAPIKey apiKey: String) -> ConfigurationBuilder {
+        return ConfigurationBuilder(withAPIKey: apiKey)
+    }
+
+    /// The Builder for ```Configuration```.
+    @objc(RCConfigurationBuilder) public class ConfigurationBuilder: NSObject {
+
+        private(set) var apiKey: String
+        private(set) var appUserID: String?
+        private(set) var observerMode: Bool = false
+        private(set) var userDefaults: UserDefaults?
+        private(set) var storeKit2Setting: StoreKit2Setting = .init(useStoreKit2IfAvailable: false)
+        private(set) var dangerousSettings: DangerousSettings?
+        private(set) var networkTimeoutSeconds: Int = Configuration.networkTimeoutSecondsDefault
+        private(set) var storeKit1TimeoutSeconds: Int = Configuration.storeKitTimeoutSecondsDefault
+
+        override public init() {
+            fatalError("Use init(withAPIKey:).")
+        }
+
+        /// Create a new builder with your API key.
+        @objc public init(withAPIKey apiKey: String) {
+            self.apiKey = apiKey
+        }
+
+        /// Update your API key.
+        @objc public func with(apiKey: String) -> ConfigurationBuilder {
+            self.apiKey = apiKey
+            return self
+        }
+
+        /// Set or update an `appUserID`.
+        @objc public func with(appUserID: String) -> ConfigurationBuilder {
+            self.appUserID = appUserID
+            return self
+        }
+
+        /// Set or update `observerMode`.
+        @objc public func with(observerMode: Bool) -> ConfigurationBuilder {
+            self.observerMode = observerMode
```

Comment:
The `or update` part here seems like it might need more clarification - you really shouldn't be updating this value after the SDK has been set. 
Since we're storing these properties in Purchases.swift, the values will be stored so if you try to update the configuration after configuration it'll be ignored. 
Maybe we could add that to the docstring?

Reply:
Ohhh, right, I didn't realize that. I'll update.

Reply:
Ok, `Set or update` is meant to be `Set or update default <property value> from default`
The builder is only used to initialize the SDK but some values might already have defaults, like the `userDefaults` so if you set it, you're actually updating the config. This is super pedantic and I'll just update them all to be `set`.

Reply:
✅